cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -1417,6 +1486,13 @@ class device_row_comparator {
     {
     }
 
+    template <typename Element, CUDF_ENABLE_IF(std::is_void_v<Element>)>
+    __device__ bool operator()(size_type const lhs_element_index,
+                               size_type const rhs_element_index) const noexcept
+    {
+      return false;
```

Comment:
I want to express here that this path is unreachable since for types which are typedef-ed to void in the Id to type map are known to be not present in the row. But for some reason, which I don't know why yet, if I instead use `CUDF_UNREACHABLE(...)` the register usage increases to 54, instead of the expected 46 in the mixed_semi_join case for integer keys and values

Reply:
A naive question: have you quantified the performance impact by reducing the register usage from 54 to 46?

Reply:
I haven't done this yet. this would be different for GPU and for joins as well. For e.g on T4, looking at the occupancy calculator, I dont expect any change in perf.

Reply:
Got it. Thanks for the prompt reply. We can proceed with the current and revisit this when wrapping up the PR, if it reduces register usage but has no performance impact, I'm inclined to use `CUDF_UNREACHABLE` for better readability.